File: 1482-how-many-numbers-are-smaller-than-the-current-number/how-many-numbers-are-smaller-than-the-current-number.cpp
class Solution {
public:
    vector<int> smallerNumbersThanCurrent(vector<int>& nums) {
        vector<int> count(101,0);
        for(auto i : nums)
        count[i]++;

        for(int i=1;i<101;i++)
        count[i] +=count[i-1];
        vector<int> res;
        for(auto i: nums)
        {
            if(i==0)
            res.push_back(0);
            else
            res.push_back(count[i-1]);
        }
        return res;
    }
};

File: 1482-how-many-numbers-are-smaller-than-the-current-number/how-many-numbers-are-smaller-than-the-current-number.py
class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:

        count = [0]*101
        for i in nums:
            count[i]+=1

        for i in range(1,101):
            count[i] +=count[i-1]
        res = []
        for i in nums:
            if i == 0:
                res.append(0)
            else:
                res.append(count[i-1])
        return res
    
